### PR TITLE
DAOS-10605 test: improve container/boundary.py

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: container_boundary
Test-repeat: 3
Skip-unit-tests: true
Skip-fault-injection-test: true

- Create pools in parallel
- Create container in parallel with a single ThreadManager
  - Reduces max number of inflight Threads
- Decrease timeout
- Get params from yaml once in setup instead of per thread
- Misc cleanup

Signed-off-by: Dalton Bohning <dalton.bohning@intel.com>

### CODE REVIEW

**Comment by daltonbohning** [comment]
The `test_2` variant passed 5 times in under 20 minutes: https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-9027/15/tests

All 3 variants passed 3 times each in under 20 minutes:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-9027/16/tests

File: src/tests/ftest/container/boundary.py [hunk 1]
```diff
@@ -1,145 +1,139 @@
...
+            TestPool: the created test pool object.
+
+        """
+        pool = self.get_pool()
+        self.pool.append(pool)
+        return pool
+
+    def create_container_and_test(self, pool=None, cont_num=1):
         """To create single container on pool.
+
```
Comment (shimizukko) [inline src/tests/ftest/container/boundary.py:58]: While you're here, you might want to update line 60 to:
`pool (TestPool): ...`

**Review by shimizukko — APPROVED** [review]

File: src/tests/ftest/container/boundary.py (hunk 1 above)
Comment (daltonbohning) [reply src/tests/ftest/container/boundary.py:58]: Thanks. I'll update if I need to push again

**Comment by daltonbohning** [comment]
ping reviewers

**Review by dinghwah — APPROVED** [review]

**Comment by ashleypittman** [comment]
Needs merging with master so the PR is up-to-date.

**Comment by daltonbohning** [comment]
Merged with master and tests still passed: https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-9027/17/pipeline

**Pull request merged**